src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -400,52 +423,30 @@ private ListBoxModel getEnvironmentIdItems(Secret formApiKey, boolean disableSsl
                 }
 
                 return items;
-            } catch (IOException | MablSystemError e) {
+            } catch (IOException | MablSystemException e) {
                 LOGGER.warning("Failed to retrieve environment IDs: " + e.getLocalizedMessage());
             }
 
             return getSelectValidApiKeyListBoxModel();
         }
 
-        public ListBoxModel doFillLabelsItems(@QueryParameter String restApiKeyId, @QueryParameter boolean disableSslVerification) {
-            if (StringUtils.isBlank(restApiKeyId)) {
-                return getSelectValidApiKeyListBoxModel();
-            }
-
-            Secret secretKey = getRestApiSecret(restApiKeyId);
-            return secretKey != null ? getLabelsItems(secretKey, disableSslVerification) : new ListBoxModel();
-        }
-
-        private ListBoxModel getLabelsItems(Secret formApiKey, boolean disableSslVerification) {
-            final MablRestApiClient client = new MablRestApiClientImpl(
-                    MABL_REST_API_BASE_URL, formApiKey, MABL_APP_BASE_URL, disableSslVerification);
-            try {
-                GetApiKeyResult apiKeyResult = client.getApiKeyResult(formApiKey);
-                if (apiKeyResult == null) {
-                   return getSelectValidApiKeyListBoxModel();
-                }
-
-                ListBoxModel items = new ListBoxModel();
-                String organizationId = apiKeyResult.organization_id;
-                GetLabelsResult labelsResult = client.getLabelsResult(organizationId);
-
-                for (GetLabelsResult.Label label : labelsResult.labels) {
-                    items.add(label.name, label.name);
-                }
-
-                return items;
-            } catch (IOException | MablSystemError e) {
-                LOGGER.warning("Failed to retrieve plan labels: " + e.getLocalizedMessage());
-            }
-
-            return getSelectValidApiKeyListBoxModel();
-        }
-
         private static ListBoxModel getSelectValidApiKeyListBoxModel() {
             final ListBoxModel listBoxModel = new ListBoxModel();
             listBoxModel.add("Select a valid API key");
             return listBoxModel;
         }
+
+        private static MablRestApiClient createMablRestApiClient(
+                final Secret key,
+                final boolean disableSslVerification,
+                final String apiBaseUrl,
+                final String appBaseUrl) {
+            return new MablRestApiClientImpl(
+                    StringUtils.isEmpty(apiBaseUrl) ? DEFAULT_MABL_API_BASE_URL : apiBaseUrl,
```

Comment:
Minor; Optionals can provide a handle interface for this behavior:
```suggestion
                    ofNullable(apiBaseUrl).orElse(DEFAULT_MABL_API_BASE_URL),
```